### PR TITLE
Fix debug output for 3D loop closure error.

### PR DESCRIPTION
**Pull request by wohe**
PAIR=spielawa

### CODE REVIEW

File: cartographer/mapping_3d/pose_graph/constraint_builder.cc [hunk 1]
```diff
@@ -244,8 +244,8 @@ void ConstraintBuilder::ComputeConstraint(
     if (match_full_submap) {
       info << " matches";
     } else {
-      const transform::Rigid3d difference = global_submap_pose.inverse() *
-                                            global_node_pose *
+      const transform::Rigid3d difference = global_node_pose.inverse() *
+                                            global_submap_pose *
                                             constraint_transform;
```
Comment (gaschler) [inline cartographer/mapping_3d/pose_graph/constraint_builder.cc]: drive-by comment: the comment that explains constraint_transform is ~50 lines above its declaration

**Comment by wohe** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user wohe. Merge queue now has a length of 1.
